source/common/umf_pools/disjoint_pool_config_parser.cpp

```diff
@@ -177,46 +177,31 @@ DisjointPoolAllConfigs parseDisjointPoolConfig(const std::string &config,
     };
 
     size_t MaxSize = (std::numeric_limits<size_t>::max)();
+    size_t EnableBuffers = 1;
 
     // Update pool settings if specified in environment.
-    size_t EnableBuffers = 1;
-    if (config != "") {
-        std::string Params = config;
-        size_t Pos = Params.find(';');
-        if (Pos != std::string::npos) {
-            if (Pos > 0) {
-                GetValue(Params, Pos, EnableBuffers);
-            }
-            Params.erase(0, Pos + 1);
-            size_t Pos = Params.find(';');
-            if (Pos != std::string::npos) {
-                if (Pos > 0) {
-                    GetValue(Params, Pos, MaxSize);
-                }
-                Params.erase(0, Pos + 1);
-                do {
-                    size_t Pos = Params.find(';');
-                    if (Pos != std::string::npos) {
-                        if (Pos > 0) {
-                            std::string MemParams = Params.substr(0, Pos);
-                            MemTypeParser(MemParams);
-                        }
-                        Params.erase(0, Pos + 1);
-                        if (Params.size() == 0) {
-                            break;
-                        }
-                    } else {
-                        MemTypeParser(Params);
-                        break;
-                    }
-                } while (true);
-            } else {
-                // set MaxPoolSize for all configs
-                GetValue(Params, Params.size(), MaxSize);
-            }
+    bool EnableBuffersSet = false;
+    bool MaxSizeSet = false;
+    size_t Start = 0;
+    size_t End = config.find(';');
+    while (true) {
```

Comment:
We actually already have a generic parser for env variables here: https://github.com/oneapi-src/unified-runtime/blob/main/source/common/ur_util.hpp#L212C33-L212C46

Can we use it? We could replace the entire logic above as well. Probably we would need to make a version of getenv_to_vec that supports ';' or just accepts strings instead of env var names and we would do the splitting here.

Reply:
I thought about it, but in SYCL_PI_LEVEL_ZERO_USM_ALLOCATOR we got a mix of named and unnamed params.
So neither getenv_to_map nor getenv_to_vec will work without changes. It would be better to change the definition of env var instead, but this PR is just a simple fix.

Reply:
The only unnamed param is the first option (EnableBuffers) - we should be able to apply getenv_to_map on the rest of the string, right?

Reply:
I know that currently getenv_to_map  accepts a name of the env var, but it should be fairly simple to refactor out a function that accepts a string instead.

Reply:
we have two unnamed params - there is also MaxPoolSize

Reply:
ah right, but it would still be fairly easy to extract the string to pass to getenv_to_map (or a string-accepting) version - just find a ';' followed by a letter (if any).